core/src/main/java/org/mskcc/oncotree/crosswalk/MSKConceptCache.java

```diff
@@ -41,6 +42,8 @@
 
     private final static Logger logger = Logger.getLogger(MSKConceptCache.class);
     private static HashMap<String, MSKConcept> oncoTreeCodesToMSKConcepts = new HashMap<String, MSKConcept>();
+    // use this to store and look up previous oncoTree codes
+    private static HashMap<String, HashSet<String>> topBraidURIsToOncotreeCodes = new HashMap<String, HashSet<String>>();
```

Comment:
a global cache is added here to hold all mappings of oncotree concept URIs (topbraid Identifiers) to the oncotree codes they might refer to.